### PR TITLE
Viet user profile

### PR DESCRIPTION
**Pull request by viethphamvn**

### CODE REVIEW

File: app/src/main/java/edu/temple/templetag/UserSettingActivity.java [hunk 1]
```diff
@@ -1,55 +1,113 @@
...
+import com.google.firebase.firestore.QueryDocumentSnapshot;
+import com.google.firebase.firestore.QuerySnapshot;
+import com.rengwuxian.materialedittext.MaterialEditText;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import edu.temple.templetag.fragments.TagListFragment;
+
 public class UserSettingActivity extends AppCompatActivity {
```
Comment (dilloncoffman) [inline app/src/main/java/edu/temple/templetag/UserSettingActivity.java]: Could we refactor the name of this activity and all its instances to be UserProfileActivity to match the UML?

File: app/src/main/java/edu/temple/templetag/UserSettingActivity.java [hunk 2]
```diff
@@ -1,55 +1,113 @@
...
+                    }
+                });
+                builder.show();
+            }
+        });
 
-        public SettingsFragment(String txt_username, String txt_email){
-            this.txt_email = txt_email;
-            this.txt_username = txt_username;
+        TagListFragment fragment = (TagListFragment) getSupportFragmentManager().findFragmentByTag("taglist");
```
Comment (dilloncoffman) [inline app/src/main/java/edu/temple/templetag/UserSettingActivity.java]: Could we rename TagListFragment and all its instances to be TagRecyclerViewFragment like lab 2 to match the UML? We'll also need a TagRecyclerViewAdapter class but not sure how far you went into implementing that in this PR.

File: app/src/main/java/edu/temple/templetag/fragments/TagListFragment.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+import android.view.ViewGroup;
+
+import edu.temple.templetag.R;
+
+/**
+ * A simple {@link Fragment} subclass.
+ * Use the {@link TagListFragment#newInstance} factory method to
+ * create an instance of this fragment.
+ */
+public class TagListFragment extends Fragment {
```
Comment (dilloncoffman) [inline app/src/main/java/edu/temple/templetag/fragments/TagListFragment.java]: Rename to TagRecyclerViewFragment and add a TagRecyclerViewAdapter like in the UML if possible

File: app/src/main/res/layout/fragment_tag_list.xml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
+<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".fragments.TagListFragment"
```
Comment (dilloncoffman) [inline app/src/main/res/layout/fragment_tag_list.xml]: We'll want to update to be fragment_tag_recycler_view and change the tools:context=".fragments.TagListFragment" to tools:context=".fragments.TagRecyclerViewFragment"

File: app/src/main/java/edu/temple/templetag/UserSettingActivity.java [hunk 5]
```diff
@@ -1,55 +1,113 @@
...
+        if (fragment != null){
+            getSupportFragmentManager().beginTransaction()
+                    .remove(fragment)
+                    .add(R.id.tagListContainer, TagListFragment.newInstance("Hello", "From UserSetting"), "taglist")
```
Comment (dilloncoffman) [inline app/src/main/java/edu/temple/templetag/UserSettingActivity.java]: Update Activity and View names to match UML

File: app/src/main/java/edu/temple/templetag/SignUpActivity.java [hunk 6]
```diff
@@ -100,22 +100,20 @@ public void onFailure(@NonNull Exception e) {
...
-                                }
-                            })
-                            .addOnFailureListener(new OnFailureListener() {
-                                @Override
-                                public void onFailure(@NonNull Exception e) {
-                                    Toast.makeText(SignUpActivity.this, "Something went wrong! Please try again.", Toast.LENGTH_LONG).show();
-                                    finish();
-                                }
-                            });
+                    firestore.collection("Users").document(firebaseUser.getUid())
```
Comment (dilloncoffman) [inline app/src/main/java/edu/temple/templetag/SignUpActivity.java:103]: How is setting different from adding a user in Firestore?

**Review by dilloncoffman — CHANGES_REQUESTED** [review]

File: app/src/main/java/edu/temple/templetag/fragments/TagListFragment.java (hunk 3 above)
Comment (viethphamvn) [reply app/src/main/java/edu/temple/templetag/fragments/TagListFragment.java]: I have not worked on it yet. I have to coordinate with Zach cuz he assigns this task to himself.

**Pull request closed without merging**